src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -50,28 +50,55 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    val negated = -rational
+    negated.numer == -rational.numer && negated.denom == rational.denom
+
   }
 
   property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+    val result = left + right
+    val expectedNumer = left.numer * right.denom + right.numer * left.denom
+    val expectedDenom = left.denom * right.denom
+    val (simplifiedNumer, simplifiedDenom) = simplifyFraction(expectedNumer, expectedDenom)
+    result.numer == simplifiedNumer && result.denom == simplifiedDenom
+
   }
 
   property("subtraction") = forAll { (left: Rational, right: Rational) =>
-    ???
+    val result = left - right
+    val expectedNumer = left.numer * right.denom - right.numer * left.denom
+    val expectedDenom = left.denom * right.denom
+    val (simplifiedNumer, simplifiedDenom) = simplifyFraction(expectedNumer, expectedDenom)
+    result.numer == simplifiedNumer && result.denom == simplifiedDenom
+
   }
 
   property("multiplication") = forAll { (left: Rational, right: Rational) =>
-    ???
+    val result = left * right
+    val expectedNumer = left.numer * right.numer
+    val expectedDenom = left.denom * right.denom
+    val (simplifiedNumer, simplifiedDenom) = simplifyFraction(expectedNumer, expectedDenom)
+    result.numer == simplifiedNumer && result.denom == simplifiedDenom
+
   }
 
   property("division") = forAll { (left: Rational, numer: Int, denom: Int) =>
     val right = Rational(if numer == 0 then 1 else numer, abs(denom) + 1)
```

Comment:
The tests must be as plain as possible (no `if-else` conditions).
Let's a bit improve it.
Please review `arbitraries` in `topic`. We already have generators for positive, negative integers and zero. Use them to be sure conditions on `numer` and `denom` are met.
Please split it on two separate tests:
1) division by positive `Rational` (could be constructed using `PosistiveInteger` arbitrary from `topic`);
1) division by negative `Rational` (could be constructed using `NegativeInteger` arbitrary from `topic`).
That's the way how you 100% avoid division by zero and check to different flows.